Dockerfile.rhel8

```diff
@@ -1,10 +1,10 @@
-FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.20-openshift-4.15 AS builder
+FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.20-openshift-4.15 AS builder
```

Comment:
should fix the filename to align(or drop the version from the filename entirely)  (which i'm sure will ripple elsewhere also)

Reply:
The ripples extended into the openshift/release repo and I remember putting in this filename to some of the forms for adding new images.

While alignment would be nice, this PR gets the actual bases correct.  Can we take the debt and merge?